code/API_definitions/Application_Endpoint_Discovery_API.yaml

```diff
@@ -0,0 +1,609 @@
+          description: |
+            When the API Consumer includes the "X-Correlator" header in the
+            request, the API provider must include it in the response with
+            the same value that was used in the request. Otherwise, it is
+            optional to include the "X-Correlator" header in the response with
+             any valid value. Recommendation is to use UUID for values.
+          schema:
+            type: string
+            format: uuid
+      responses:
+        "200":
+          description: |
+            Successful response, returning the closest App Instance Endpoint
+            Zone to the user device identified in the request header.
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/AccessEndpoints"
+        "400":
+          $ref: "#/components/responses/400BadRequest"
+        "401":
+          $ref: "#/components/responses/401Unauthorized"
+        "403":
+          $ref: "#/components/responses/403Forbidden"
+        "404":
+          $ref: "#/components/responses/404NotFound"
+        "405":
+          $ref: "#/components/responses/405MethodNotAllowed"
+        "406":
+          $ref: "#/components/responses/406Unacceptable"
+        "429":
+          $ref: "#/components/responses/429TooManyRequests"
+        "500":
+          $ref: "#/components/responses/500InternalServerError"
+        "502":
+          $ref: "#/components/responses/502BadGateway"
+        "503":
+          $ref: "#/components/responses/503ServiceUnavailable"
+        "504":
+          $ref: "#/components/responses/504GatewayTimeout"
+      tags:
+        - Discovery
+      summary: |
+        Returns the Endpoint of the closest App Instance to user device
+        identified in the request.
+      description: |
+        On receiving this request, the network will return the name
+        of the App Instance with the shortest network path to the end user
+        device identified in the request.
+components:
+  securitySchemes:
+    openId:
+      description: OpenID Provider Configuration Information.
+      type: openIdConnect
+      openIdConnectUrl: .well-known/openid-configuration
+  headers:
+    X-Correlator:
+      description: |
+       When the API Consumer includes the "X-Correlator" header in the request,
+        the API provider must include it in the response with the same value t
+        hat was used in the request. Otherwise, it is optional to include the
+        "X-Correlator" header in the response with any valid value.
+        Recommendation is to use UUID for values.
+      required: false
+      schema:
+        type: string
+        format: uuid
+  schemas:
+    AppId:
+      type: string
+      format: uuid
+      description: A globally unique identifier associated
+        with the application.
+        This GUID will have been previously provided by the
+        Edge Cloud Zone provider.
+    AccessEndpoints:
+      type: object
```

Comment:
I think one comment is lost for some reasons. From the last comment "May be an application is composed of multiple components and offers one or more endpoints to its clients for different purposes. In that case data model should consider those possibilities and accordingly consider them in defining the AccessEndpoint structure e.g. one or more ports, fqdns etc as an example."

In my opinion the AccessEndpoint should be an array of objects instead of a single object as it is defined now to have one or more endpoints. And each endpoint may have a qualifier defined by API invoker to help differentiate between the endpoints based on the purpose of the endpoint in client side code. May be such qualifiers can be provided during app onboarding time of the application along with the components to which the endpoint belongs to in the /app-endpoints response.

Reply:
Agree with you and as it was concluded in the edge cloud meeting for this version we keep a single endpoint